app/components/MapSwitcher.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faLayerGroup } from "@fortawesome/free-solid-svg-icons";
+import type { Map } from "maplibre-gl";
+
+interface Props {
+  map: Map | undefined;
+}
+
+const MapSwitcher = ({ map }: Props) => {
+  const [activeLayer, setActiveLayer] = useState<string>("custom");
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const toggleLayer = (layerName: string) => {
+    if (!map) return;
+
+    const layers = ['google-satellite-layer', 'usgs-topo-layer', 'atlmaps-layer'];
+
+    layers.forEach((layer) => {
+      map.setLayoutProperty(layer, 'visibility', 'none');
+    });
+
+    if (layerName !== 'custom') {
+      map.setLayoutProperty(`${layerName}-layer`, 'visibility', 'visible');
+    }
+
+    setActiveLayer(layerName);
+    setIsDropdownOpen(false);
+  };
+
+  return (
+    <div className="absolute top-4 right-4">
+      <button
+        onClick={() => setIsDropdownOpen(!isDropdownOpen)}
+        className="bg-white p-2 rounded-md shadow-md"
+      >
+        <FontAwesomeIcon icon={faLayerGroup} />
+        <span className="sr-only">Toggle layer menu</span>
+      </button>
+      {isDropdownOpen && (
+        <div className="absolute right-0 mt-2 w-48 rounded-md shadow-lg bg-white ring-1 ring-black ring-opacity-5">
+          <div className="py-1" role="menu" aria-orientation="vertical" aria-labelledby="options-menu">
+            <button
```

Comment:
See my other comment about creating an array of objects for the layers. With that in place, you should do something like
~~~~jsx
{mapLayers.map((layer) => {
  return (
    <button
      onClick={() => toggleLayer(layer)}
      ....
    >
      {layer.label}
    </button>
})}